core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineSqlite.java

```diff
@@ -0,0 +1,241 @@
+package com.scalar.db.storage.jdbc;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.DataType;
+import com.scalar.db.storage.jdbc.query.InsertOnConflictDoUpdateQuery;
+import com.scalar.db.storage.jdbc.query.SelectQuery;
+import com.scalar.db.storage.jdbc.query.SelectWithLimitQuery;
+import com.scalar.db.storage.jdbc.query.UpsertQuery;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A RdnEngineStrategy implementation for SQLite.
+ *
+ * <p>Namespace: Added to table prefix like `${namespace}_${tableName}`.
```

Comment:
With this convention, the following case could happen?
- there are 2 namespaces: `a` and `a_b`
- namespace `a` contains table `b_c`
- namespace `a_b` contains table `c`
- the two different physical table names conflict...? (`a_b_c`)

Reply:
I noticed SQLite accepts some special characters like `$`. How about using those special unusual character as a delimiter like this?
```
sqlite> create table ns$tbl (id text);
sqlite> select * from ns$tbl;
sqlite> insert into ns$tbl values ('hello');
sqlite> select * from ns$tbl;
hello
```

Reply:
It would be great ScalarDB itself rejects namespace name and table name that include those special characters.

Reply:
@komamitsu Good point. I agree with you in that people usually do not choose `$` for table names, while `_` is frequently used.
I will use `$` for the separator in SQLite implementation.

> It would be great ScalarDB itself rejects namespace name and table name that include those special characters.

I also agree with idea.

I'll try to reject it in SQLite storage layer in order not to affect other storage implementations.

Reply:
- change namespace separator: 79681c88ac3616278b4f4a4051181fe351943618
- add validation: 8db587ae4b2c68142f1f50d6fee48237fe775dd9